### PR TITLE
Add shellcheck compilation for alpine, and classic installation for debian

### LINKED ISSUE #17: Add Shellcheck for .sh linting
It could be interesting to add shellcheck to this image to support `.sh` linting.

https://github.com/koalaman/shellcheck

### PR DESCRIPTION
**Pull request by henri9813**
Fixes #17

Added shellcheck by compilation for alpine and classical install (apt) for debian.

### CODE REVIEW

File: base-alpine.Dockerfile [hunk 1]
```diff
@@ -51,3 +51,10 @@ RUN cd /opt \
     && tar xzf yarn.tar.gz -C yarn --strip-components 1 \
     && cd /usr/local/bin \
     && ln -s /opt/yarn/bin/yarn
+
+# Install Shellcheck
```
Comment (soullivaneuh) [inline base-alpine.Dockerfile]: Shellcheck setup

File: base-alpine.Dockerfile [hunk 2]
```diff
@@ -51,3 +51,10 @@ RUN cd /opt \
...
+RUN export scversion="stable" \
+	&& wget "https://storage.googleapis.com/shellcheck/shellcheck-${scversion}.linux.x86_64.tar.xz" \
+	&& tar --xz -xvf shellcheck-"${scversion}".linux.x86_64.tar.xz \
```
Comment (soullivaneuh) [inline base-alpine.Dockerfile]: Please use long options.

File: base-alpine.Dockerfile [hunk 3]
```diff
@@ -51,3 +51,10 @@ RUN cd /opt \
     && tar xzf yarn.tar.gz -C yarn --strip-components 1 \
     && cd /usr/local/bin \
     && ln -s /opt/yarn/bin/yarn
+
+# Install Shellcheck
+RUN export scversion="stable" \
```
Comment (soullivaneuh) [inline base-alpine.Dockerfile]: No need for an export here, we will always use stable version.

File: base-alpine.Dockerfile [hunk 4]
```diff
@@ -51,3 +51,10 @@ RUN cd /opt \
...
+	&& wget "https://storage.googleapis.com/shellcheck/shellcheck-${scversion}.linux.x86_64.tar.xz" \
+	&& tar --xz -xvf shellcheck-"${scversion}".linux.x86_64.tar.xz \
+	&& cp shellcheck-"${scversion}"/shellcheck /usr/bin/ \
+	&& shellcheck --version
```
Comment (soullivaneuh) [inline base-alpine.Dockerfile]: Missing new line.

File: base-alpine.Dockerfile [hunk 5]
```diff
@@ -51,3 +51,10 @@ RUN cd /opt \
     && tar xzf yarn.tar.gz -C yarn --strip-components 1 \
     && cd /usr/local/bin \
     && ln -s /opt/yarn/bin/yarn
+
+# Install Shellcheck
+RUN export scversion="stable" \
+	&& wget "https://storage.googleapis.com/shellcheck/shellcheck-${scversion}.linux.x86_64.tar.xz" \
+	&& tar --xz -xvf shellcheck-"${scversion}".linux.x86_64.tar.xz \
+	&& cp shellcheck-"${scversion}"/shellcheck /usr/bin/ \
```
Comment (soullivaneuh) [inline base-alpine.Dockerfile]: Please cleanup all the not needed file after the setup.

**Review by soullivaneuh — CHANGES_REQUESTED** [review]

File: base-alpine.Dockerfile [hunk 6]
```diff
@@ -51,3 +51,12 @@ RUN cd /opt \
     && tar xzf yarn.tar.gz -C yarn --strip-components 1 \
     && cd /usr/local/bin \
     && ln -s /opt/yarn/bin/yarn
+
+# Shellcheck setup
+ADD https://storage.googleapis.com/shellcheck/shellcheck-stable.linux.x86_64.tar.xz /opt/shellcheck.tar.xz
+RUN cd /opt \
+	&& mkdir shellcheck \
+	&& tar --xz --extract --file shellcheck.tar.xz --directory shellcheck \
```
Comment (soullivaneuh) [inline base-alpine.Dockerfile:59]: - Please use long options.
- `-x` and `--extract` are the same options.

File: base-alpine.Dockerfile [hunk 7]
```diff
@@ -51,3 +51,12 @@ RUN cd /opt \
...
+	&& cp shellcheck/shellcheck-stable/shellcheck /usr/bin/ \
+	&& shellcheck --version \
+	&& rm -rf shellcheck shellcheck.tar.xz
```
Comment (soullivaneuh) [inline base-alpine.Dockerfile:62]: Please use long options.

**Review by soullivaneuh — CHANGES_REQUESTED** [review]

File: base-alpine.Dockerfile (hunk 7 above)
Comment (henri9813) [reply base-alpine.Dockerfile:62]: --xz is to specify that this archive is `xz` format

Comment (henri9813) [reply base-alpine.Dockerfile:62]: rm only know parameter: -r, -R and -f not --recursive and --force

**Review by soullivaneuh — APPROVED** [review]

**Comment by soullivaneuh** [comment]
Thank you @henri9813.

**Pull request merged**